src/PactNet.Native/Scenarios.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using PactNet.Models;
+
+namespace PactNet.Native
+{
+    /// <summary>
+    /// Defines the scenarios static list for messaging support
+    /// </summary>
+    public class Scenarios
+    {
+        /// <summary>
+        /// The available scenarios
+        /// </summary>
+        internal readonly List<Scenario> AllScenarios = new List<Scenario>();
+
+        /// <summary>
+        /// Add a scenario
+        /// </summary>
+        /// <param name="scenario">the scenario to add</param>
+        public void AddScenario(Scenario scenario)
+        {
+            if (scenario == null)
+            {
+                throw new ArgumentNullException(nameof(scenario));
+            }
+
+            if (Exist(scenario.Description))
+            {
+                throw new InvalidOperationException($"Scenario \"{scenario.Description}\" already added");
+            }
+
+            AllScenarios.Add(scenario);
+        }
+
+        /// <summary>
+        /// Add multiple scenarios
+        /// </summary>
+        /// <param name="scenarios">the scenario list to add</param>
+        public void AddScenarios(IReadOnlyCollection<Scenario> scenarios)
+        {
+            if (scenarios == null || scenarios.Any() == false)
```

Comment:
It's idiomatic to use `!scenarios.Any()` instead of `scenarios.Any() == false`

Reply:
I don't understand.
The code is scenarios.Any() == false.
Is there another way you are hinting at?